tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Test_Tov.cpp

```diff
@@ -0,0 +1,135 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Tov.hpp"
+#include "PointwiseFunctions/EquationsOfState/EquationOfState.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Literals.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+// IWYU pragma: no_forward_declare EquationsOfState::EquationOfState
+
+namespace {
+
+constexpr double polytropic_constant = 8.0;
+double expected_newtonian_radius() noexcept {
+  /* Analytic solution in Newtonian limit for polytropic TOV integration has
+  $R = \pi / \alpha$ ; $M = 4\pi^{2}\rho_{0c} / \alpha^{3}$
+   Here $\rho_{0c}$ is the central mass density and $\alpha = sqrt(2\pi/K)$
+   with $K$ the polytropic constant
+  */
+  return M_PI / sqrt(2.0 * M_PI / polytropic_constant);
+}
+
+double expected_newtonian_mass(const double central_mass_density) noexcept {
+  return 4.0 * central_mass_density * square(M_PI) /
+         (cube(sqrt(2.0 * M_PI / polytropic_constant)));
+}
+
+void test_tov(const std::unique_ptr<EquationsOfState::EquationOfState<true, 1>>&
+                  equation_of_state,
+              const double central_mass_density, const size_t num_pts,
+              const size_t current_iteration,
+              const bool newtonian_limit) noexcept {
+  Approx custom_approx = Approx::custom().epsilon(1.0e-08).scale(1.0);
+  const double initial_log_enthalpy =
+      std::log(get(equation_of_state->specific_enthalpy_from_density(
+          Scalar<double>{central_mass_density})));
+  const double surface_log_enthalpy = 0.0;
+  const double step = (surface_log_enthalpy - initial_log_enthalpy) / num_pts;
+  const double final_log_enthalpy =
+      initial_log_enthalpy + (current_iteration + 1) * step;
+  const EinsteinSolutions::TovSolution tov_out_full(
+      equation_of_state, central_mass_density, surface_log_enthalpy);
+  const EinsteinSolutions::TovSolution tov_out_test(
```

Comment:
Let's rename `_test` to `_intermediate` like you have below, that's much clearer. I think a small comment like `// Integrate only to some intermediate value, not the surface. Then compare to interpolated values.`

I would also move this down to where it's first used, line 73. The general rule in C++ is to not declare anything before it is used. This means as late is possible.